pkg/kudoctl/packages/resolver/resolver_incluster.go

```diff
@@ -0,0 +1,43 @@
+package resolver
+
+import (
+	"fmt"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/convert"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+)
+
+// InClusterResolver resolves packages that are already installed in the cluster on the client-side. Note, that unlike
+// other resolvers, the resulting 'packages.Package' struct does not contain package 'packages.Files' (we don't have
+// the original files).
+type InClusterResolver struct {
+	c  *kudo.Client
+	ns string
+}
+
+func (r InClusterResolver) Resolve(name string, appVersion string, operatorVersion string) (*packages.Package, error) {
+	ovn := kudoapi.OperatorVersionName(name, operatorVersion)
```

Comment:
Why, oh why, do we create an operatorVersionName just from the operator name and the operatorVersion? Shouldn't we have the appVersion in the OV-Name as well??? Not that this is an issue on this PR, I just noticed it here...

Reply:
In theory, a bump in the `appVersion` should lead to a bump in `operatorVersion` so that using latter in the OV name isn't necessary. But otherwise, "cause it was always this way" 🤷

Reply:
No - `appVersion` and `operatorVersion` are independent. For example, Kafka was just bumped with a new `appVersion`, but the operator didn't get any new features or changes it has the same `operatorVersion`...

Reply:
> but the operator didn't get any new features or changes it has the same operatorVersion

How can it be the same operator if the _underlying app has changed?_ Even minor app changes can bring arbitrary changes in APIs, behavior, etc. I'm not sure if that was a good idea.

Reply:
It's not the same operator. That's why the `appVersion` was changed. An Operator can only be uniqly identified by OV-Version **and** AppVersion. That's why we have both in the file name in the repository.

https://github.com/kudobuilder/kudo/blob/main/keps/0019-package-api-versioning.md#example-hypothetical-timeline-for-apache-cassandra-releases

Reply:
Hm, besides the fact that [appVersion is optional](https://github.com/kudobuilder/kudo/blob/main/pkg/kudoctl/packages/types.go#L92), the `OperatorVersion` resource currently does not include it in the name 🤷

Reply:
Yeah, i know. That's what irritated me. Seems we just didn't implement the change, it's mentioned in the KEP: https://github.com/kudobuilder/kudo/blob/main/keps/0019-package-api-versioning.md#naming-of-operatorversion-objects

I've created an issue for it, but no idea how to solve it yet.

Reply:
historically it was expected that a bump in `appVersion` would bump `operatorVersion`.   They are broken apart so that an operator can evolve independently without an appVersion change...   exposing the appVersion was important because that is what the end user is interested in and the early days of kudo didn't expose that... just the opVersion which was frustrating from an end user stand point.  IMO it is mistake to bump appVersion without bumping opVersion.  This was a point of debate for the versioning kep.. as long as we use the entire combined version I guess we will be fine.